gammapy/maps/wcs.py

```diff
@@ -21,6 +21,28 @@
     'WcsGeom',
 ]
 
+def _check_width(width):
+    """Convert input width as tuple of float to pass to WcsGeom.create"""
+    if isinstance(width, (Angle,Quantity)):
+        if width.size>1:
+            return tuple([_.to('deg').value for _ in width])
+        else:
+            return (width.to('deg').value, width.to('deg').value)
+    elif np.isscalar(width):
+        return (float(width), float(width))
+    elif isinstance(width,(list,tuple)):
+        if np.isscalar(width[0]):
+            if len(width)==1:
+                return (float(width[0]),float(width[0]))
+            else:
+                return tuple(width)
+        elif isinstance(width[0], (Angle,Quantity)):
+            return tuple([_.to('deg').value for _ in width])
+        else:
+            raise TypeError("Unsupported input for width")
```

Comment:
If you want to be kind, you show the user what they passed in, i.e. write this:
```
raise TypeError('Unsupported width: {!r}'.format(width))
```

Reply:
Done